Shared/Samples/Densify and generalize geometry/DensifyAndGeneralizeGeometryView.swift

```diff
@@ -0,0 +1,184 @@
+        
+        /// The graphic for displaying the points of the resultant geometry.
+        let resultPointsGraphic: Graphic = {
+            let symbol = SimpleMarkerSymbol(style: .circle, color: .magenta, size: 7)
+            return Graphic(symbol: symbol)
+        }()
+        
+        /// The graphic for displaying the lines of the resultant geometry.
+        let resultPolylineGraphic: Graphic = {
+            let symbol = SimpleLineSymbol(style: .solid, color: .magenta, width: 3)
+            return Graphic(symbol: symbol)
+        }()
+        
+        /// A mutable point collection from which the original polyline and
+        /// mulitpoint geometries are made.
+        private let pointCollection: MutablePointCollection = {
+            // The spatial reference for the sample.
+            let spatialReference = SpatialReference(wkid: WKID(rawValue: 32126)!)
+            var collection = MutablePointCollection(spatialReference: spatialReference)
+            collection.append(contentsOf: [
+                Point(x: 2330611.130549, y: 202360.002957),
+                Point(x: 2330583.834672, y: 202525.984012),
+                Point(x: 2330574.164902, y: 202691.488009),
+                Point(x: 2330689.292623, y: 203170.045888),
+                Point(x: 2330696.773344, y: 203317.495798),
+                Point(x: 2330691.419723, y: 203380.917080),
+                Point(x: 2330435.065296, y: 203816.662457),
+                Point(x: 2330369.500800, y: 204329.861789),
+                Point(x: 2330400.929891, y: 204712.129673),
+                Point(x: 2330484.300447, y: 204927.797132),
+                Point(x: 2330514.469919, y: 205000.792463),
+                Point(x: 2330638.099138, y: 205271.601116),
+                Point(x: 2330725.315888, y: 205631.231308),
+                Point(x: 2330755.640702, y: 206433.354860),
+                Point(x: 2330680.644719, y: 206660.240923),
+                Point(x: 2330386.957926, y: 207340.947204),
+                Point(x: 2330485.861737, y: 207742.298501)
+            ])
+            return collection
+        }()
+        
+        /// A Boolean indicating whether to generalize.
+        @Published var shouldGeneralize = false
+        
+        /// The max deviation for generalization.
+        @Published var maxDeviation = 10.0
+        
+        /// A Boolean indicating whether to densify.
+        @Published var shouldDensify = false
+        
+        /// The max segment length for densifying.
+        @Published var maxSegmentLength = 100.0
+        
+        /// Reset the model values to the originals.
+        func reset() {
+            shouldDensify = false
+            maxDeviation = 10.0
+            shouldGeneralize = false
+            maxSegmentLength = 100.0
+            updateGraphics()
+        }
+        
+        /// Update the result polyline and mulitpoint graphics after densifying
+        /// and generalizing.
+        func updateGraphics() {
```

Comment:
It may not follow the design, but I felt like generalize and densify should be separated into 2 methods, and the UI controls should be mutually exclusive. Currently it is "first generalize then densify" if both toggles are on. I'll leave it open for other reviewer's thoughts.

Reply:
FWIW, I agree.

Reply:
I tried to implement it in two different methods, and I am not sure it is the right approach. Since each method only either densifies or generalizes and then updates the graphics accordingly, they each throw away the result of the other. That's great if that's what you want, but want to see the results combined, it would be much less complex and make more sense to have them in the same method since work they on the same polyline. Let me know what you guys think though.